test/src/test/java/org/corfudb/integration/StreamingPatternsIT.java

```diff
@@ -0,0 +1,577 @@
+package org.corfudb.integration;
+
+import com.google.common.reflect.TypeToken;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import org.corfudb.protocols.wireprotocol.Token;
+import org.corfudb.runtime.CorfuStoreMetadata.Timestamp;
+import org.corfudb.runtime.MultiCheckpointWriter;
+import org.corfudb.runtime.collections.CorfuRecord;
+import org.corfudb.runtime.collections.CorfuStore;
+import org.corfudb.runtime.collections.CorfuStoreEntry;
+import org.corfudb.runtime.collections.CorfuStreamEntries;
+import org.corfudb.runtime.collections.CorfuStreamEntry;
+import org.corfudb.runtime.collections.CorfuTable;
+import org.corfudb.runtime.collections.StreamListenerResumeOrDefault;
+import org.corfudb.runtime.collections.StreamListenerResumeOrFullSync;
+import org.corfudb.runtime.collections.Table;
+import org.corfudb.runtime.collections.TableOptions;
+import org.corfudb.runtime.collections.TxnContext;
+import org.corfudb.runtime.view.TableRegistry;
+import org.corfudb.test.SampleSchema;
+import org.corfudb.test.SampleSchema.Uuid;
+import org.corfudb.test.SampleSchema.SampleTableAMsg;
+import org.corfudb.test.SampleSchema.SampleTableBMsg;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+
+/**
+ * Simple test that inserts data into CorfuStore and tests Streaming.
+ */
+@Slf4j
+public class StreamingPatternsIT extends AbstractIT {
+
+    private static String corfuSingleNodeHost;
+    private static int corfuStringNodePort;
+    private static String singleNodeEndpoint;
+    private Process corfuServer;
+    private CorfuStore store;
+
+    private final String namespace = "UT-namespace";
+    private final String defaultTableName = "table_default";
+    private final String defaultTag = "sample_streamer_1";
+
+    private final int sleepTime = 100;
+
+    private Table<SampleSchema.Uuid, SampleTableAMsg, SampleSchema.Uuid> tableDefault;
+
+    /**
+     * Load properties for a single node corfu server before each test
+     */
+    @Before
+    public void loadProperties() {
+        corfuSingleNodeHost = PROPERTIES.getProperty("corfuSingleNodeHost");
+        corfuStringNodePort = Integer.valueOf(PROPERTIES.getProperty("corfuSingleNodePort"));
+        singleNodeEndpoint = String.format("%s:%d",
+                corfuSingleNodeHost,
+                corfuStringNodePort);
+    }
+
+    /**
+     * Test stream listener implementation with default re-subscription policy, i.e.,
+     * resume subscription if possible or subscribe to latest point in the log when not possible.
+     */
+    private class StreamListenerResumeOrDefaultImpl extends StreamListenerResumeOrDefault {
```

Comment:
StreamListenerResumeOrDefault talks about a possible data loss. Do you think its a good idea to add an IT to demonstrate such possibility?

Reply:
There is, take a look at testDefaultPortionOfResumeOrDefaultPolicy (we exercise the Data loss part as well)

Reply:
Thanks